cmake/modules/extensions.cmake

```diff
@@ -1559,6 +1559,66 @@ function(zephyr_syscall_header_ifdef feature_toggle)
   endif()
 endfunction()
 
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a warning/
+# fatal error message is printed (depends on REQUIRED flag).
+# Paths are relative to the current module's zephyr/blobs directory.
+#
+# Usage:
+#   zephyr_blobs_verify([PATHS path [paths...]]
+#                       [REQUIRED])
+#
+# Example:
+# zephyr_blobs_verify(REQUIRED)           # verify all blobs and fail on error
+# zephyr_blobs_verify(PATHS img/file.bin) # verify a single path
+function(zephyr_blobs_verify)
+  if(NOT DEFINED ZEPHYR_CURRENT_MODULE_NAME)
```

Comment:
Module should be caller specified, caller may choose to use `MODULE ${CMAKE_CURRENT_MODULE_NAME}` when calling the function from inside its own CMake glue code.